test/CQRS/LeanCode.CQRS.MassTransitRelay.Tests/Middleware/EventsPublisherMiddlewareTests.cs

```diff
@@ -0,0 +1,194 @@
+#nullable enable
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using FluentAssertions;
+using LeanCode.Contracts;
+using LeanCode.CQRS.Execution;
+using LeanCode.CQRS.MassTransitRelay.Middleware;
+using LeanCode.DomainModels.Model;
+using LeanCode.IntegrationTestHelpers;
+using MassTransit;
+using MassTransit.Testing;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace LeanCode.CQRS.MassTransitRelay.Tests.Middleware;
+
+public sealed class EventsPublisherMiddlewareTests : IDisposable, IAsyncLifetime
+{
+    private readonly IHost host;
+    private readonly TestServer server;
+    private readonly IServiceProvider serviceProvider;
+    private readonly ITestHarness harness;
+
+    public EventsPublisherMiddlewareTests()
+    {
+        host = new HostBuilder()
+            .ConfigureWebHost(webHost =>
+            {
+                webHost
+                    .UseTestServer()
+                    .ConfigureServices(cfg =>
+                    {
+                        cfg.AddAuthentication().AddTestAuthenticationHandler();
+                        cfg.AddMassTransitTestHarness(c =>
+                        {
+                            c.AddConsumer(typeof(TestEventConsumer), typeof(TestEventConsumerDefinition));
+                        });
+                        cfg.AddAsyncEventsInterceptor();
+                    })
+                    .Configure(app =>
+                    {
+                        app.UseAuthentication();
+                        app.UseMiddleware<EventsPublisherMiddleware>();
+                        app.Run(ctx =>
+                        {
+                            var payload = ctx.GetCQRSRequestPayload();
+                            DomainEvents.Raise(new TestEvent1());
+                            payload.SetResult(Execution.ExecutionResult.WithPayload(CommandResult.Success));
+                            return Task.CompletedTask;
+                        });
+                    });
+            })
+            .Build();
+
+        server = host.GetTestServer();
+        serviceProvider = server.Services;
+        harness = serviceProvider.GetRequiredService<ITestHarness>();
+
+        harness.TestTimeout = TimeSpan.FromSeconds(1);
+    }
+
+    [Fact]
+    public async Task Does_not_propagate_anything_when_the_user_is_unauthorized()
+    {
+        await harness.Start();
+        var ctx = await SendAsync(new TestCommand(), null);
+        AssertCommandResultSuccess(ctx);
```

Comment:
Does this assertion add anything? The test is about events, not command execution.